private/pkg/storage/bucket.go

```diff
@@ -61,7 +93,7 @@ type WriteBucket interface {
 	// The returned WriteObjectCloser is not thread-safe.
 	//
 	// Returns error on system error.
-	Put(ctx context.Context, path string) (WriteObjectCloser, error)
+	Put(ctx context.Context, path string, opts ...PutOption) (WriteObjectCloser, error)
```

Comment:
This is a breaking change, so before this PR is merged, all implementations will need to have PRs ready to be updated to this